test/integration/targets/known_hosts/tasks/main.yml

```
@@ -61,6 +61,27 @@
     - 'known_hosts.stdout_lines[5].startswith("# example.net")'
     - 'known_hosts.stdout_lines[-1].strip() == example_org_rsa_key.strip()'
 
+# https://github.com/ansible/ansible/issues/84952
+# test addition when ~/.ssh/ directory missing and path=default
+- name: move ~/.ssh/ directory away
+  command: mv ~/.ssh/ ~/.ssh_bak/
```

The test suite doesn't explicitly create ~/.ssh, and mv will fail if it doesn't exist.

```suggestion
- name: Check if ~/.ssh exists
  stat:
    path: ~/.ssh
  register: default_ssh_dir

- name: move ~/.ssh/ directory away
  command: mv ~/.ssh/ ~/.ssh_bak/
  when: default_ssh_dir.stat.exists
  register: backup_default_ssh_dir
```